src/ClientBase.php

```diff
@@ -108,14 +108,14 @@ public function post($endpoint, array $headers = [], $payload)
 
     protected function setCredentials(array $config = [])
     {
-        if (!isset($config['username']) || !isset($config['password'])) {
-            $message = 'A username and a password are required for authenticated to the Canada Post API.';
+        if (!isset($config['username']) || !isset($config['password']) || !isset($config['customer_number'])) {
+            $message = 'A username, password and customer number are required for authenticated to the Canada Post API.';
             throw new \InvalidArgumentException($message);
         }
 
         $this->username = $config['username'];
         $this->password = $config['password'];
-        $this->customerNumber = $config['customerNumber'];
+        $this->customerNumber = $config['customer_number'];
```

Comment:
The customer number is actually not a requirement and there are several endpoints that can be called without it. However, let's leave it like this right now so we can make a release and we can improve on this later. Each endpoint should be validating its required arguments.